### DISCUSSION TITLE
export table content from PBIX

### DISCUSSION THREAD

**Discussion by cyrilbrd** [DaxStudio/DaxStudio#1181] [General]
[probably related to this post.](https://github.com/DaxStudio/DaxStudio/discussions/1166#discussion-5531515)
Hi,
After posting an issue in Power BI Forum pertaining to an ongoing bug limiting export to 300 rows, ([link](https://community.fabric.microsoft.com/t5/Desktop/Data-exceeds-the-limit-300-Rows/m-p/3437360#M1140278))
I was advised by a user a give DaxStudio a try as it would allow me to extract data from tables created in PBD.
I am so far unable to fully appreciate the extend of what DaxStudio can do.
I have not been able to date to do a proper extraction using this App, 
May I ask if someone could give me a hint of where to look?
Given that I would like to be able to select an existing pbix, and download the info based on existing filters.
This is the advice given to me by user @foodd (maybe in this forum?)

"_USING DAX STUDIO TO EXPORT FROM VISUAL:  DAX Studio when installed will appear as an External Tool in Power BI Desktop.  One of it's many features allows the export of data from selected Visual to CSV or Excel (leverage created DAX to export).

In Power BI Desktop using Performance analyzer you are able to capture and reveal the DAX query behind any Visual on a Report.  Then, copy the DAX query, and use this in DAX Studio to return the data behind the visual.  This is also quite handy when writing documentation._"

Would this work with matrix and tables?
I get an output that is not what I require.
Thanks in advance for any input.

**Comment by marcosqlbi** [comment]
You can capture the query of any visual, but you should modify it in order to get all the rows that you want. The queries generated by Power BI use TOPN function to limit the rows returned in order to not "block" the report. If you manipulate the query raising or removing the limit, then you can get as many rows as you want.

**Comment by dgosbell** [comment]
You can use the [copy query](https://susheelaakulu.medium.com/what-is-performance-analyzer-in-powerbi-893d1a065cd3#:~:text=You%20can%20copy%20the%20query%20in%20the%20clipboard,the%20Query%20Plan%20and%20Server%20Timings%20tracing%20options.) feature in the Power BI Desktop Performance Analyzer pane to capture the query from a visual.

Or you could use the [Query Builder](https://daxstudio.org/docs/features/query-builder/) in DAX Studio to create your own query.

Then you would change the [output](https://daxstudio.org/docs/features/output-modes/) to file and when you run your query you will get prompted for a name for your csv file.
